client/my-sites/themes/theme-options.js

```diff
@@ -36,27 +36,29 @@ const purchase = config.isEnabled( 'upgrades/checkout' )
 		} ),
 		header: i18n.translate( 'Purchase on:', {
 			context: 'verb',
-			comment: 'label for selecting a site for which to purchase a theme'
+			comment: 'label for selecting a siteId for which to purchase a theme'
 		} ),
 		getUrl: getPurchaseUrl,
-		hideForTheme: ( state, theme, site ) => ! theme.price || isActive( state, theme.id, site ) || isPurchased( state, theme.id, site )
+		hideForTheme: ( state, theme, siteId ) =>
+			! theme.price || isActive( state, theme.id, siteId ) || isPurchased( state, theme.id, siteId )
 	}
 	: {};
 
 const activate = {
 	label: i18n.translate( 'Activate' ),
-	header: i18n.translate( 'Activate on:', { comment: 'label for selecting a site on which to activate a theme' } ),
+	header: i18n.translate( 'Activate on:', { comment: 'label for selecting a siteId on which to activate a theme' } ),
```

Comment:
I'm afraid there are a couple more `header` strings that were messed up because of the `site` -> `siteId` replacement...